health/health.d/postgres.conf

```diff
@@ -14,3 +14,215 @@ component: PostgreSQL
+    delay: down 15m multiplier 1.5 max 1h
+     info: average cache hit ratio in db $label:database table $label:table over the last minute
+       to: dba
+
+ template: postgres_table_index_cache_io_ratio
+       on: postgres.table_index_cache_io_ratio
+    class: Workload
+     type: Database
+component: PostgreSQL
+    hosts: *
+   lookup: average -1m unaligned of miss
+     calc: 100 - $this
+    units: %
+    every: 1m
+     warn: $this < (($status >= $WARNING)  ? (70) : (60))
+     crit: $this < (($status == $CRITICAL) ? (60) : (50))
+    delay: down 15m multiplier 1.5 max 1h
+     info: average index cache hit ratio in db $label:database table $label:table over the last minute
+       to: dba
+
+ template: postgres_table_toast_cache_io_ratio
+       on: postgres.table_toast_cache_io_ratio
+    class: Workload
+     type: Database
+component: PostgreSQL
+    hosts: *
+   lookup: average -1m unaligned of miss
+     calc: 100 - $this
+    units: %
+    every: 1m
+     warn: $this < (($status >= $WARNING)  ? (70) : (60))
+     crit: $this < (($status == $CRITICAL) ? (60) : (50))
+    delay: down 15m multiplier 1.5 max 1h
+     info: average TOAST hit ratio in db $label:database table $label:table over the last minute
+       to: dba
+
+ template: postgres_table_toast_index_cache_io_ratio
+       on: postgres.table_toast_index_cache_io_ratio
+    class: Workload
+     type: Database
+component: PostgreSQL
+    hosts: *
+   lookup: average -1m unaligned of miss
+     calc: 100 - $this
+    units: %
+    every: 1m
+     warn: $this < (($status >= $WARNING)  ? (70) : (60))
+     crit: $this < (($status == $CRITICAL) ? (60) : (50))
+    delay: down 15m multiplier 1.5 max 1h
+     info: average index TOAST hit ratio in db $label:database table $label:table over the last minute
+       to: dba
+
+ template: postgres_table_rows_dead_ratio
+       on: postgres.table_rows_dead_ratio
+    class: Errors
+     type: Database
+component: PostgreSQL
+    hosts: *
+     calc: $dead
+    units: %
+    every: 1m
+     warn: $this > (($status >= $WARNING)  ? (10) : (15))
+    delay: down 15m multiplier 1.5 max 1h
+     info: dead rows percentage in db $label:database table $label:table
+       to: dba
+
+ template: postgres_table_bloat_size_perc
+       on: postgres.table_bloat_size_perc
+    class: Errors
+     type: Database
+component: PostgreSQL
+    hosts: *
+     calc: $bloat
+    units: %
+    every: 1m
+     warn: $this > (($status >= $WARNING)  ? (60) : (70))
+     crit: $this > (($status == $CRITICAL) ? (70) : (80))
+    delay: down 15m multiplier 1.5 max 1h
+     info: bloat size percentage in db $label:database table $label:table
+       to: dba
+
+ template: postgres_table_last_autovacuum_time
+       on: postgres.table_autovacuum_since_time
+    class: Errors
+     type: Database
+component: PostgreSQL
+    hosts: *
+     calc: $time
+    units: seconds
+    every: 1m
+     warn: $this != nan AND $this > 60 * 60 * 24 * 7
```

Comment:
Why do we need one week to raise this alert? Probably half week is safe enough to report the issue.

Reply:
Why do you think half week is better?

---

I am not Postgres DBA and can't really say how often vacuum/analyze is supposed to run. I used [this alarm](https://awesome-prometheus-alerts.grep.to/rules#rule-postgresql-1-4) as an example.

Reply:
I expect vacuum to run on database in a short period of time, because if we expect a long time to run, it will need more [resources](https://wiki.postgresql.org/wiki/Tuning_Your_PostgreSQL_Server). Of course I do not expect to run every hour, but I think every week can be too long for a big server.

Reply:
@anayrat, hey. Can you suggest a value for "time elapsed since last auto(vacuum|analyze)"?

Reply:
Such threshold will really depend on your workload :/ 
It is hard to give a value for any database.

Reply:
Ok, does it make sense to have this kind of alarm by default with a high number like I did (7 days)?

~ rationale:
 - 7 days is a lot and likely there is a problem.
 - not a small period of time, so no false positives.
 - people can use this stock and modify it to their needs.

If it is completely garbage 😄 I can remove it.

Reply:
@ilyam8 I understand the alert is useful, but as @anayrat said the workload determines. I already worked with a database that if we only run `vacuum` in the seventh day, I would need hours to run.

Reply:
I think it's good to have an alert with a higher threshold - gives users an easy way to configure a threshold that suits them without spamming them with easily triggering alerts. 

Also easier for user to update a threshold vs creating the alert from scratch.